### PR TITLE
Added GA API version for Scheduled Query Rule

### PR DESCRIPTION
**Pull request by abversqr**
Added GA API version for Scheduled Query Rule

This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by fearthecowboy** [comment]
Yoink! @sarangan12 -- I'm assigning this to me so I can use it to train more reviewers.  You get a freebie!

**Comment by fearthecowboy** [comment]
Hey @abversqr  -- it appears you have a few validation errors in your spec -- can you correct these and update the PR? 

https://travis-ci.org/Azure/azure-rest-api-specs/jobs/365583431#L736

If you want to run the linter yourself locally, [install autorest](https://aka.ms/autorest/install) and try:

``` batch
# from the folder where the readme.md file is
autorest --azure-validator readme.md 
```

It looks mostly like one typo (`date=time`?) missing examples and an API that should be defined.

**Comment by fearthecowboy** [comment]
Hey @abversqr  -- this looks like a new API is being added -- has this been reviewed by the API review board? 

If not, you should schedule a one hour SKYPE meeting with Garrett Serack, David Justice, Johan Stenberg, Gaurav Bhatnagar, Darrel Miller and Niklas Gustafsson . Preferably Tuesday afternoon sometime between 1 and 5pm.

File: specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json
Comment (fearthecowboy) [inline specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json]: You shouldn't have `x-ms-code-generation-settings` in the actual swagger file (I don't think we even use this anymore at all).

File: specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json
Comment (fearthecowboy) [inline specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json]: The `description` should  document the API (the presence of the swagger file is telling you how you should physically call this method). You can drop everything after `Creates or updates an log search rule.`

File: specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json
Comment (fearthecowboy) [inline specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json]: This should use `x-ms-examples` and the example data should be in a separate file. (and then the linter won't yell at you for missing examples :D )

File: specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json
Comment (fearthecowboy) [inline specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json]: again, fix the `operationId` to match the pattern (ie`ScheduledQueryRules_Get`)

File: specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json
Comment (fearthecowboy) [inline specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json]: again, move example into external file with `x-ms-examples`

File: specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json
Comment (fearthecowboy) [inline specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json]: You should support a `"default"` response for errors. (see example https://github.com/abversqr/azure-rest-api-specs/blob/2e4c790d16eb6be9299db7a75798bca1697fd5e9/specification/monitor/resource-manager/microsoft.insights/stable/2018-03-01/metricAlert_API.json#L503 )

File: specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json
Comment (fearthecowboy) [inline specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json]: Error states should be modeled as `"default"` responses and have a model to deserialize the error into.

A lot of these things should have been caught in the API review board -- I'll pause the review here, and wait until after the review board has seen these APIs.

**Review by fearthecowboy — CHANGES_REQUESTED** [review]
You will need to resolve a bunch of this before I can continue the review -- and the API review board should sign off.

File: specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json
Comment (ravbhatnagar) [inline specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json]: this should also be readonly since the name comes from URL

File: specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json
Comment (ravbhatnagar) [inline specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json]: SKUs of a resource is indicated through the ARM top level "sku" property. The details can be found in the ARM resource provider contract

File: specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json
Comment (ravbhatnagar) [inline specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json]: please define as an enum and list all the possible values.

File: specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json
Comment (abversqr) [reply specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json]: Fixed.

File: specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json
Comment (abversqr) [reply specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json]: Fixed in latest iteration

File: specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json
Comment (abversqr) [reply specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json]: Fixed in latest iteration

File: specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json
Comment (abversqr) [reply specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json]: Fixed in latest iteration

File: specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json
Comment (abversqr) [reply specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json]: Fixed in latest iteration

File: specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json
Comment (abversqr) [reply specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json]: Fixed in latest iteration

**Comment by ravbhatnagar** [comment]
Looks good!

**Comment by abversqr** [comment]
Hey @fearthecowboy, I have resolved all comments. Can ou please have a look into this and sign-off if no blocking changes are pending,

File: specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json
Comment (fearthecowboy) [inline specification/monitor/resource-manager/microsoft.insights/stable/2018-04-16/scheduledQueryRule_API.json]: You should have a `default` response modeled here -- if the service fails for any reason, the client needs to be able to get the error response.

**Review by fearthecowboy — APPROVED** [review]
LGTM.

There is still a R3026 Linter violation -- the RP will fix in a subsequent PR : __"we are working on this, but will take couple of weeks to get this done. Once this done, we will push this change in new PR for review. Let us know if you see any concern with this."__

**Pull request merged**
